packages/core/types/src/types/core/attributes/blocks.ts

```diff
@@ -8,6 +78,15 @@ export type Blocks = Attribute.OfType<'blocks'> &
   Attribute.WritableOption &
   Attribute.VisibleOption;
 
-export type BlocksValue<T extends object = object> = T;
+export type BlocksValue = RootNode[];
+
+// Type utils needed for the blocks renderer and the blocks editor (useBlocksStore)
+export type BlockNode<NodeKind extends 'root' | 'inline' | 'all'> = NodeKind extends 'root'
```

Comment:
I would reco you create the InlineNodes | AllNodes types to to make this generic simpler. 

Can you share an example of how you expect to use this generic ?

Reply:
I'm using it there: https://github.com/strapi/strapi/pull/18671/files#diff-3fd35ca8f33b0853b875bbacefa504addeb66debfb797cd51862486225902e70R93

My plan for it was initially to set the slate Descendant type to `BlockNode<'inline'>`. But it doesn't look like I can, as Slate considers a top level node a descendant of editor. I'll double check but it looks like I could just Block all nodes then (without the generic)